nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestCommentsMigrator.java

```diff
@@ -233,6 +236,48 @@ public void testMigrationFromPropertyToSecuredWithCommentsNotCorrectlyFilled() {
             assertTrue(caughtEventMessages.contains(message));
             assertTrue(session.exists(new IdRef(docId)));
         }
+
+    }
+
+    /**
+     * Test the migration in the case where the {@code comment:parentId} is not empty, but it doesn't exist.
+     */
+    @Test
+    @LogCaptureFeature.FilterOn(logLevel = "WARN")
+    @Deploy("org.nuxeo.ecm.platform.comment.tests:OSGI-INF/disable-removing-comment-children-listener.xml")
+    public void testMigrationFromPropertyToSecuredWithRemovedParentComment() {
+        // NB_COMMENTS_BY_FILE*4 + NB_COMMENT_TO_REPLY_ON_IT*NB_REPLY_BY_COMMENT (5 levels of reply) = 200 + 50
+        createCommentsAsProperty(false, true);
+
+        // Second step of migrate: from 'Property' to 'Secured'
+        migrateFromPropertyToSecured(new CommentsMigrator(), false, true);
+        transactionalFeature.nextTransaction();
+
+        List<LogEvent> events = logCaptureResult.getCaughtEvents();
+        assertEquals(NB_COMMENTS_BY_FILE + 1, events.size());
+
+        String query = String.format("SELECT %s FROM Comment WHERE %s = '%s'", ECM_UUID, COMMENT_PARENT_ID,
+                fileToCommentAndRemove.getId());
+        List<String> unMigratedCommentDocIds = session.queryProjection(query, 0, 0)
+                                                      .stream()
+                                                      .map(m -> (String) m.get(ECM_UUID))
+                                                      .collect(Collectors.toList());
+
+        // The comments not migrated should be there but, not under 'fileToCommentAndRemove'
+        assertEquals(NB_COMMENTS_BY_FILE, unMigratedCommentDocIds.size());
+        List<String> caughtEventMessages = logCaptureResult.getCaughtEventMessages();
+
+        assertTrue(caughtEventMessages.contains(String.format(
+                "Some comments have not been migrated, see logs for more information. The folder containing these comments will be renamed to %s",
+                UNMIGRATED_COMMENTS_FOLDER_NAME)));
+
+        for (String docId : unMigratedCommentDocIds) {
+            String message = String.format(
+                    "The comment document model with IdRef: %s cannot be migrated, because its parent: %s cannot be found",
+                    docId, fileToCommentAndRemove.getId());
+            assertTrue(caughtEventMessages.contains(message));
```

Comment:
You assert everything has been caught (the 51 warns) but you don't assert there are only 51 (no duplicates, erronous messages). Maybe that would be overkill ? I leave it to you !

Reply:
My idea here is to be sure that all my expected things are here not on the whole process, i mean if a new warn can be logged by another process (for example once who deploy, others services...) i will not be impacted by it as the main message will be logged as `WARN`. this a first thing in other hand have look [here](https://github.com/nuxeo/nuxeo/blob/fix-NXP-28620-Fix-Comment-Migration-failure/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestCommentsMigrator.java#L486) i check that the `50` comments will be not migrated

Reply:
My question is about asserting the number of warns generated introduced by your changes is equal and not superior to the number of comments not migrated, I'm not talking about checking the effective comments migrated as you already assert you have the right warn for each one.

It's up to you as I said !

Reply:
yes this check is already done see [here](https://github.com/nuxeo/nuxeo/blob/fix-NXP-28620-Fix-Comment-Migration-failure/nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/TestCommentsMigrator.java#L267) this check ensure that at least i have the warn messages of my comments but to check the inverse i don't think is really needed